### PR TITLE
Scala 2.12

### PR DESCRIPTION
**Pull request by jhnsmth**
Changes made in this PR:
-  pure webjar is downgraded. Because it looks like 2.83 is a [different library](https://github.com/pure/pure) 
- changes related to ammonite-ops reflect the changes made [here](https://github.com/lihaoyi/Ammonite/commit/fe29bc906ad2f931f0fdfe673afe53e8b2dfd762)
- scala 2.10 is removed, as compilation of scalatex-api failed. (I doubt it was ever compiled before)
- some of the test in ErrorTests are separated for 2.11 and 2.12,  as the style of error messages has been changed in 2.12.

### CODE REVIEW

**Comment by jhnsmth** [comment]
Ready for review!

**Comment by romainreuillon** [comment]
Nice !

File: api/src/test/scala-2.12/scalatex/Scala212ErrorTests.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import scalatex.Internals.{DebugFailure, twRuntimeErrors, twfRuntimeErrors}
+import ErrorTests.check
+
+object Scala212ErrorTests extends TestSuite {
+  val tests = TestSuite {
+    'chained {
+      'curlies {
+        * - check(
+          twRuntimeErrors("@p{@Seq(1, 2, 3).foldLeft(0)}"),
+          "type mismatch",
```
Comment (jhnsmth) [inline api/src/test/scala-2.12/scalatex/Scala212ErrorTests.scala]: I have some concerns about this test. On scala 2.11 it throws "missing argument list for method foldLeft", as expected, but on scala 2.12 it complains about type mismatch. If remove `@p{...}` it produce a correct error (about missing arguments) and `@p{@Seq(1, 2, 3).foldLeft(0)(_ + _)}` throws no error, so I think the behaviour is correct, but anyway it worth to give it some attention.

**Comment by lihaoyi** [comment]
Thanks! Traveling now but will take a look later

**Comment by romainreuillon** [comment]
Any news on merging this?

**Comment by lihaoyi** [comment]
been traveling and on the road since scala exchange; will merge this, but
havent had a good contiuous block of time with a computer since then. Will
try to schedule some time to properly review before end of year but can't
make promises

**Comment by lihaoyi** [comment]
Tested locally, looks good to me

**Comment by lihaoyi** [comment]
Published as 0.3.7

**Comment by romainreuillon** [comment]
That's great. Thx!

**Pull request merged**